src/v/raft/consensus.cc

```diff
@@ -1568,13 +1568,14 @@ consensus::disk_append(model::record_batch_reader&& reader) {
               update_follower_stats(configurations.back().cfg);
               f = _configuration_manager.add(std::move(configurations));
           }
+
           return f.then([this, ret = ret] {
-              return _configuration_manager
-                .maybe_store_highest_known_offset(
-                  ret.last_offset, ret.byte_size)
-                .then([ret = ret]() mutable {
-                    return ss::make_ready_future<ret_t>(std::move(ret));
+              (void)ss::with_gate(
+                _bg, [this, last_offset = ret.last_offset, sz = ret.byte_size] {
```

Comment:
You don't handle gate exceptions here

Reply:
I will add handle_exception

Reply:
I've end up adding check if gate is closed.